### PR TITLE
Add IERC transferFrom to HTS Precompile Acceptance Tests 

### LINKED ISSUE #497: Add IERC transferFrom to htsPrecompile approval acceptance tests
### Problem

Currently the `htsPrecompile.spec.ts` has tests for approval flow.
However, it doesn't follow up with actual transfer tests.

### Solution

Add an additional step to each `approval` test to test the `transferFrom()` call.
Ensure both `IERC20` and `IERC721` get coverage
    - `IERC20(token).transferFrom(sender, recipient, amount);`
    - `IERC721(token).transferFrom(from, to, tokenId);`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by georgi-l95**
Signed-off-by: georgi-l95 <glazarov95@gmail.com>

**Description**:
Adds `ERC721Mock.json` .
Ensure both `IERC20` and `IERC721` function `transferFrom` gets coverage.
```
- IERC20(token).transferFrom(sender, recipient, amount);
- IERC721(token).transferFrom(from, to, tokenId);
```

**Related issue(s)**:

Fixes #497 

**Notes for reviewer**:
Also adds coverage for:

```
- approveNFT(address token, address spender, uint256 tokenId)
- IERC721(token).balanceOf(address owner);
- IERC721(token).getApproved(uint256 tokenId);
```

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/server/tests/acceptance/htsPrecompile.spec.ts [hunk 1]
```diff
@@ -178,11 +181,20 @@ describe('HTS Precompile Acceptance Tests', async function () {
       const { responseCode } = txBeforeReceipt.events.filter(e => e.event === 'ResponseCode')[0].args;
       expect(responseCode).to.equal(TX_SUCCESS_CODE);
   
-      await baseHTSContract.approvePublic(HTSTokenContractAddress, accounts[2].wallet.address, amount, { gasLimit: 1_000_000 });
-  
-      const txAfter = await baseHTSContract.allowancePublic(HTSTokenContractAddress, BaseHTSContractAddress, accounts[2].wallet.address);
+      await baseHTSContract.approvePublic(HTSTokenContractAddress, accounts[0].wallet.address, amount, { gasLimit: 1_000_000 });
```
Comment (Nana-EC) [inline packages/server/tests/acceptance/htsPrecompile.spec.ts:184]: Is this testing the right flow.
Initially I believe it was right.
`account[0]` approved an allowance of `amount` for `account[2]`
This update is approving for oneself and then sending to someone else, which should work as `accounts[0]` is the treasury no?

Comment (Nana-EC) [inline packages/server/tests/acceptance/htsPrecompile.spec.ts:184]:
```suggestion
      await baseHTSContract.approvePublic(HTSTokenContractAddress, accounts[2].wallet.address, amount, { gasLimit: 1_000_000 });
```

File: packages/server/tests/acceptance/htsPrecompile.spec.ts [hunk 2]
```diff
@@ -178,11 +181,20 @@ describe('HTS Precompile Acceptance Tests', async function () {
...
+
+      // grant KYC
+      const grantKycTx = await baseHTSContractOwner.grantTokenKycPublic(HTSTokenContractAddress, accounts[2].wallet.address, { gasLimit: 1_000_000 });
```
Comment (Nana-EC) [inline packages/server/tests/acceptance/htsPrecompile.spec.ts:187]:
```suggestion
      const grantKycTx = await baseHTSContractOwner.grantTokenKycPublic(HTSTokenContractAddress, accounts[1].wallet.address, { gasLimit: 1_000_000 });
```

File: packages/server/tests/acceptance/htsPrecompile.spec.ts [hunk 3]
```diff
@@ -178,11 +181,20 @@ describe('HTS Precompile Acceptance Tests', async function () {
...
+      const responseCodeGrantKyc = (await grantKycTx.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode;
+      expect(responseCodeGrantKyc).to.equal(TX_SUCCESS_CODE);
+
+      //transfer with transferFrom
+      await HTSTokenContract.transferFrom(accounts[0].wallet.address, accounts[2].wallet.address, amount);
```
Comment (Nana-EC) [inline packages/server/tests/acceptance/htsPrecompile.spec.ts:192]:
```suggestion
      await HTSTokenContract.transferFrom(accounts[2].wallet.address, accounts[1].wallet.address, amount);
```

File: packages/server/tests/acceptance/htsPrecompile.spec.ts [hunk 4]
```diff
@@ -214,6 +226,22 @@ describe('HTS Precompile Acceptance Tests', async function () {
...
+      expect(await NFTokenContract.balanceOf(accounts[0].wallet.address)).to.equal(1);
+      expect(await NFTokenContract.balanceOf(accounts[2].wallet.address)).to.equal(0);
+
+      // grant KYC
+      const grantKycTx = await baseHTSContractOwner.grantTokenKycPublic(NftHTSTokenContractAddress, accounts[2].wallet.address, { gasLimit: 1_000_000 });
+      const responseCodeGrantKyc = (await grantKycTx.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode;
+      expect(responseCodeGrantKyc).to.equal(TX_SUCCESS_CODE);
+
+      //transfer
+      await NFTokenContract.transferFrom(accounts[0].wallet.address, accounts[2].wallet.address, NftSerialNumber);
```
Comment (Nana-EC) [inline packages/server/tests/acceptance/htsPrecompile.spec.ts:240]: there was no approval prior to this transferFrom

**Review by Nana-EC — CHANGES_REQUESTED** [review]
I'm not sure the tests cases are right.
Can you confirm the right accounts are first approving and the approved accounts are then doing the sending

File: packages/server/tests/acceptance/htsPrecompile.spec.ts (hunk 1 above)
Comment (georgi-l95) [reply packages/server/tests/acceptance/htsPrecompile.spec.ts]: Yep, you're right. My flow was a wrong. I've fixed it now and left some comments for clarification.

File: packages/server/tests/acceptance/htsPrecompile.spec.ts (hunk 4 above)
Comment (georgi-l95) [reply packages/server/tests/acceptance/htsPrecompile.spec.ts]: Working on it.

**Pull request closed without merging**